reactor-core-micrometer/src/test/java/reactor/core/observability/micrometer/MicrometerObservationListenerTest.java

```diff
@@ -367,4 +369,125 @@ void observationStoppedByError() {
 			.hasLowCardinalityKeyValue("reactor.status",  "error")
 			.hasError(exception);
 	}
+
+	@Test
+	void whenPropagationPrioritizesParentFromContext() {
+		configuration = new MicrometerObservationListenerConfiguration(
+			MicrometerObservationListener.ANONYMOUS_OBSERVATION,
+			//note: "type" key is added by MicrometerObservationListenerConfiguration#fromFlux (which is tested separately)
+			KeyValues.of("testTag1", "testTagValue1","testTag2", "testTagValue2"),
+			registry,
+			false);
+
+		//detached
+		TestObservationRegistry otherRegistry = TestObservationRegistry.create();
+		Observation parent = Observation.start("testParent", otherRegistry);
+		parent.openScope();
+
+		//attached to test registry
+		Observation inRegistry = Observation.start("inregistry", registry);
+		inRegistry.openScope();
+		Context contextWithAParent = Context.of(subscriberContext).put(Micrometer.OBSERVATION_CONTEXT_KEY, parent);
+
+		//NB: registry.getCurrentObservation() always reflects the last Observation.openScope, even if Observation was started from a different registry
+		assertThat(registry.getCurrentObservation()).as("inRegistry is reported as current / ThreadLocal")
+			.isSameAs(otherRegistry.getCurrentObservation())
+			.isSameAs(inRegistry);
+
+		MicrometerObservationListener<Integer> listener = new MicrometerObservationListener<>(contextWithAParent, configuration);
+
+		listener.doFirst(); // forces observation start and discovery of the parent
+
+		assertThat(listener.originalContext).as("originalContext").isSameAs(contextWithAParent);
+		assertThat(listener.contextWithScope).as("contextWithScope")
+			.matches(c -> c.hasKey(Micrometer.OBSERVATION_CONTEXT_KEY), "has OBSERVATION_CONTEXT_KEY");
+
+		AtomicReference<Observation.Context> parentRef = new AtomicReference<>();
+		AtomicReference<Observation.Context> inRegistryRef = new AtomicReference<>();
+
+		assertThat(otherRegistry)
+			.hasSingleObservationThat()
+			.hasNameEqualTo("testParent")
+			.hasBeenStarted()
+			.isNotStopped()
+			.hasNoKeyValues()
+			.satisfies(parentRef::set);
+		assertThat(registry)
+			.hasObservationWithNameEqualTo("inregistry")
+			.that()
+			.satisfies(inRegistryRef::set);
+
+		assertThat(registry)
```

Comment:
If there should be no remaining spans in the thread I would check it, otherwise I would clear the thread local (I'd do it for all tests)

Reply:
see above, should now be done